Prevent runtime issues due to DYLD_LIBRARY_PATH being set, so use
DYLD_FALLBACK_LIBRARY_PATH instead.

$ cd /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_release_ports_java_zeroc-ice35-java/zeroc-ice35-java/work/Ice-3.5.0/java/test/Slice/generation
$ DYLD_LIBRARY_PATH=/opt/local/lib:/opt/local/lib/db46 ./run.py 
starting client... ok
dyld: Symbol not found: __cg_jpeg_resync_to_restart
  Referenced from: /System/Library/Frameworks/ApplicationServices.framework/Versions/A/Frameworks/ImageIO.framework/Versions/A/ImageIO
  Expected in: /opt/local/lib/libJPEG.dylib
 in /System/Library/Frameworks/ApplicationServices.framework/Versions/A/Frameworks/ImageIO.framework/Versions/A/ImageIO
unexpected exit status: expected: 0, got -5
Traceback (most recent call last):
  File "./run.py", line 28, in <module>
    clientProc.waitTestSuccess()
  File "/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_release_ports_java_zeroc-ice35-java/zeroc-ice35-java/work/Ice-3.5.0/scripts/Expect.py", line 564, in waitTestSuccess
    test(self.exitstatus, exitstatus)
  File "/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_release_ports_java_zeroc-ice35-java/zeroc-ice35-java/work/Ice-3.5.0/scripts/Expect.py", line 549, in test
    assert False
AssertionError


diff -ru Ice-3.5.0.orig/scripts/TestUtil.py Ice-3.5.0/scripts/TestUtil.py
--- Ice-3.5.0.orig/scripts/TestUtil.py	2013-03-11 08:19:47.000000000 -0700
+++ Ice-3.5.0/scripts/TestUtil.py	2013-08-27 18:11:26.000000000 -0700
@@ -182,7 +182,7 @@
 def dumpenv(env, lang):
     if env is None:
         env = os.environ
-    vars = ["PATH", "LD_LIBRARY_PATH", "DYLD_LIBRARY_PATH", "SHLIB_PATH", "LIBPATH", "LD_LIBRARY_PATH_64"]
+    vars = ["PATH", "LD_LIBRARY_PATH", "DYLD_LIBRARY_PATH", "DYLD_FALLBACK_LIBRARY_PATH", "SHLIB_PATH", "LIBPATH", "LD_LIBRARY_PATH_64"]
     if lang is None:
         vars.extend(["CLASSPATH", "MONO_PATH", "DEVPATH", "PYTHONPATH", "RUBYLIB"])
     elif lang == "cpp":
@@ -205,7 +205,7 @@
     if isWin32():
         addPathToEnv("PATH", libpath, env)
     elif isDarwin():
-        addPathToEnv("DYLD_LIBRARY_PATH", libpath, env)
+        addPathToEnv("DYLD_FALLBACK_LIBRARY_PATH", libpath, env)
     elif isAIX():
         addPathToEnv("LIBPATH", libpath, env)
     else:
